### PR TITLE
Fix/zios 9897 render transparent background images (png) correctly (fix for UIPasteboard)

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

This PR is a follow-up of https://github.com/wireapp/wire-ios/pull/2021.

### Issues

When copy and paste a PNG image with transparent background in Wire, the image has a solid color background in conversation.

### Causes

The image data is encoded as JPG when copying and pasting in Wire

### Solutions

Support PNG encoding/decoding for UIPasteboard with MediaAsset object.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Components/UIImage+MediaAssert.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        }
+    }
+
+    public func isGIF() -> Bool {
+        return false
+    }
+
+    public func isTransparent() -> Bool {
+        guard let alpha: CGImageAlphaInfo = self.cgImage?.alphaInfo else { return false }
+        return alpha == .first || alpha == .last || alpha == .premultipliedFirst || alpha == .premultipliedLast
```
Comment (vytis) [inline Wire-iOS/Sources/UserInterface/Components/UIImage+MediaAssert.swift]: Maybe it's better to switch over the values here and list all variants - it's very easy to miss some cases here (like `.alphaOnly`)

**Comment by vytis** [comment]
So just a question, how should the transparent images be rendered? Looking at snapshot tests in #2021 I see that they have a border around it, is that the expected behaviour? Did design look at it?

**Comment by billypchan** [comment]
@vytis The border shows or not depend on the image's size, there should be a gray hairline around the image form for the original design:

ImageMessageCell.m:
`
- (void)updateImageBorder
{
    BOOL showBorder = !self.imageSmallerThanMinimumSize;
    self.fullImageView.layer.borderWidth = showBorder ? UIScreen.hairline : 0;
    self.fullImageView.layer.borderColor = [UIColor colorWithWhite:0 alpha:0.08].CGColor;
}
`

**Comment by vytis** [comment]
It might be that the logic there was made for non-transparent images though. Can you ping @eike96 about this? In my personal opinion it would be much better to avoid adding this border to transparent images, but it's of course up to them to decide.

**Comment by billypchan** [comment]
@vytis after discussion with designer, the image border color is set depends on color variant.

**Comment by billypchan** [comment]
@vytis after discussion with designer, the image border color is set depends on color variant.

**Comment by mikeger** [comment]
So `Vox populi` asked here to remove the border, we are not removing it now? https://github.com/wireapp/wire-ios/issues/1953

**Comment by billypchan** [comment]
@mikeger I will discuss again with the designer about updated the original design for the border handling for transparent images.

**Pull request merged**